From 409a0e2fd41da7a4b04672136aa7d749988faf61 Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 25 Sep 2024 16:41:55 +0100
Subject: [PATCH 1283/1350] overlays: wm8960-soundcard: Fix clock declaration

How did this ever work? The static-clock declaration is fine, but the
reference to it is attached to part of the simple-audio-card node,
rather than the instantiation of the codec driver (a subnode of the
I2C controller).

Move the clock reference where it should be, and fix a few minor
cosmetic issues at the same time.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 .../arm/boot/dts/overlays/wm8960-soundcard-overlay.dts | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

--- a/arch/arm/boot/dts/overlays/wm8960-soundcard-overlay.dts
+++ b/arch/arm/boot/dts/overlays/wm8960-soundcard-overlay.dts
@@ -13,7 +13,7 @@
 	};
 
 	fragment@1 {
-		target-path="/";
+		target-path = "/";
 		__overlay__ {
 			wm8960_mclk: wm8960_mclk {
 				compatible = "fixed-clock";
@@ -29,17 +29,18 @@
 			#size-cells = <0>;
 			status = "okay";
 
-			wm8960: wm8960 {
+			wm8960: wm8960@1a {
 				compatible = "wlf,wm8960";
 				reg = <0x1a>;
 				#sound-dai-cells = <0>;
 				AVDD-supply = <&vdd_5v0_reg>;
 				DVDD-supply = <&vdd_3v3_reg>;
+				clocks = <&wm8960_mclk>;
+				clock-names = "mclk";
 			};
 		};
 	};
 
-
 	fragment@3 {
 		target = <&sound>;
 		slave_overlay: __overlay__ {
@@ -67,10 +68,9 @@
 			simple-audio-card,cpu {
 				sound-dai = <&i2s_clk_producer>;
 			};
+
 			dailink0_slave: simple-audio-card,codec {
 				sound-dai = <&wm8960>;
-				clocks = <&wm8960_mclk>;
-				clock-names = "mclk";
 			};
 		};
 	};
